### PR TITLE
Startup migration bug

### PR DESCRIPTION
**Pull request by chlupac**
#318 solution

### CODE REVIEW

**Comment by sbwalker** [comment]
One question I have is is related to the name of the new config section - calling it "Oqtane" seems redundant as the entire config file belongs to the Oqtane framework? Wouldn't it make more sense to name the new section "Installation" - as this section is simply replicating the fields which are usually entered via the UI. There is already an installation config setting called "SiteTemplate" which could be moved to this section ( see SiteRepository.cs for _config.GetSection("SiteTemplate") ).

  "Installation": {
    "HostUser": "",
    "HostPassword": "",
    "HostEmail": "",
    "SiteTemplate": ""
  }

Note that there is another section used by the framework currently - but it is not included in the current default config file ( see LogManager.cs ):

  "Logging": {
    "LogLevel": {
      "Default": "Information"
    }
  }

**Comment by sbwalker** [comment]
we should remove the modulecreator modification from this PR as it is not related

**Comment by chlupac** [comment]
Ok, I can change appsetting.json this way, it's more descriptive, good idea.

**Comment by chlupac** [comment]
@sbwalker Done. you can merge it

File: Oqtane.Server/Extensions/PermissionExtension.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using System.Text;
+using System.Text.Json;
+using Oqtane.Models;
+
+namespace Oqtane.Extensions
+{
+    public static class PermissionExtension
+    {
+        public static string EncodePermissions(this IEnumerable<Permission> permissionList)
+        {
```
Comment (chlupac) [inline Oqtane.Server/Extensions/PermissionExtension.cs:12]: Copy from _PermissionRepository Function is static utility and for call is not necessary repository instance.
Refactored as extension.

Comment (chlupac) [reply Oqtane.Server/Extensions/PermissionExtension.cs:12]: Refactoring of all calls will be done in next step.

**Comment by sbwalker** [comment]
One comment I have is related to this code:

                   //TODO this is wrong. Decide default password / email or throw exception ??
                    Password = GetInstallationConfig(SettingKeys.HostPasswordKey, "oQtane123"),
                    Email = GetInstallationConfig(SettingKeys.HostEmailKey, "nobody@cortonso.com"),
                    DisplayName = GetInstallationConfig(SettingKeys.HostUserKey, Constants.HostUser),

I would think that that Password, Email, and DisplayName should be set to "" by default. And only if a person wants to use the "silent install" option would they specify values for these settings. Anytime an application includes a default password - especially for a highly privileged account - it has the potential to cause problems as it gives an easy attack vector for malicious users.

**Comment by chlupac** [comment]
Agree, I want to discuss this. I will check all required properties and when is not defined, system will fallback to GUI setup

**Comment by chlupac** [comment]
Default password removed, fallback to IDE

**Comment by aubrey-b** [comment]
I get the tenant is null error even after this PR.

**Comment by sbwalker** [comment]
What does your appsettings.json look like?

**Comment by chlupac** [comment]
@aubrey-b How to reproduce bug ?

**Comment by chlupac** [comment]
@aubrey-b solved in #329

**Comment by aubrey-b** [comment]
@sbwalker appsettings:

```
{
  "ConnectionStrings": {
    "DefaultConnection": "Data Source=(LocalDb)\\MSSQLLocalDB;AttachDbFilename=C:\\Users\\aubrey\\source\\repos\\oqtane.framework\\Oqtane.Server\\Data\\Oqtane-202004021802.mdf;Initial Catalog=Oqtane-202004021802;Integrated Security=SSPI;"
  },
  "Installation": {
    "DefaultAlias": "",
    "HostUser": "",
    "HostPassword": "",
    "HostEmail": "",
    "SiteTemplate": "",
    "DefaultTheme": "",
    "DefaultLayout": "",
    "DefaultContainer": ""
  }
}

```

**Comment by aubrey-b** [comment]
@chlupac Sorry to respond so late, but I just add a second site and then when I restart the application I get the error message similar to #318

**Pull request merged**